core/src/main/scala/kafka/log/Log.scala

```diff
@@ -588,7 +588,9 @@ class Log(@volatile var dir: File,
       // reset the index size of the currently active log segment to allow more entries
       activeSegment.resizeIndexes(config.maxIndexSize)
       nextOffset
```

Comment:
Here we are trusting the consistency of the log start offset and the segments in the log. If a segment is corrupt and we have to truncate some data, could the log end offset be reset to an offset that is earlier than the log end offset?

Reply:
Good point. I made some changes to handle this case as well.